[DOC-9753] Document minimum supported vCPU per node for Dedicated on Azure (#18320)
cockroachdb/docs: src/current/v22.2/recommended-production-settings.md

- For cluster stability, Cockroach Labs recommends a _minimum_ of {% include {{ page.version.version }}/prod-deployment/provision-cpu.md threshold='minimum' %}, and strongly recommends no fewer than {% include {{ page.version.version }}/prod-deployment/provision-cpu.md threshold='absolute_minimum' %} per node. In a cluster with too few CPU resources, foreground client workloads will compete with the cluster's background maintenance tasks. For more information, see [capacity planning issues](cluster-setup-troubleshooting.html#capacity-planning-issues).

    {{site.data.alerts.callout_info}}

>     Clusters deployed in CockroachDB Cloud can be created with a minimum of 2vCPUs per node on AWS and GCP or 4vcPUs per node on Azure.

Author: ```suggestion
    Clusters deployed in CockroachDB {{ site.data.products.cloud }} can be created with a minimum of 2vCPUs per node on AWS and GCP or 4vcPUs per node on Azure.
```

>     Clusters deployed in CockroachDB Cloud can be created with a minimum of 2vCPUs per node on AWS and GCP or 4vcPUs per node on Azure (Limited Access).

Author: ```suggestion
    Clusters deployed in CockroachDB Cloud can be created with a minimum of 2vCPUs per node on AWS and GCP or 4vcPUs per node on Azure.
```

>     Clusters deployed in CockroachDB {{ site.data.products.cloud }} can be created with a minimum of 2vCPUs per node on AWS and GCP or 4vcPUs per node on Azure.

Author: ```suggestion
    Clusters deployed in CockroachDB {{ site.data.products.cloud }} can be created with a minimum of 2 vCPUs per node on AWS and GCP or 4 vCPUs per node on Azure.
```
